24H Study Room/Mocks/MockFireAuthService.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  MockFireAuthService.swift
+//  24H Study Room
+//
+//  Created by Jia-Jiuan Tsai on 2022-09-05.
+//
+
+import Foundation
+
+class MockFireAuthService: FireAuthServicing {
+    func createUser(email: Email, password: Password, completion: @escaping (Result<UserCredential, Error>) -> Void) {
```

Comment:
If you use this mock in tests, the tests will all just stall, because `completion` is never called. 

A typical mock should do some hardcoded behavior when its methods are called, but it should still work. 

This is how I usually write them:

```swift
final class MockAuthService: FireAuthServicing {
    var createUserResult: Result<UserCredential, Error>
    var loginResult: Result<UserCredential, Error>
    var logoutResult: Result<UserCredential, Error>
    
    init(
        createUserResult: Result<UserCredential, Error> = .success(.mock),
        loginResult: Result<UserCredential, Error> = .success(.mock),
        logoutResult: Result<UserCredential, Error> = .success(.mock)
    ) {
        self.createUserResult = createUserResult
        self.loginResult = loginResult
        self.logoutResult = logoutResult
    }
    
    func createUser(email: Email, password: Password, completion: @escaping (Result<UserCredential, Error>) -> Void) {
        completion(createUserResult)
    }
    
    func login(email: Email, password: Password, completion: @escaping (Result<UserCredential, Error>) -> Void) {
        completion(loginResult)
    }
    
    func logout(completion: @escaping (Result<Void, Error>) -> Void) {
        completion(logoutResult)
    }
}
```